src/fhirtypes/ElementDefinition.ts

```diff
@@ -392,6 +392,14 @@ export class ElementDefinition {
           // @ts-ignore
           diff[prop] = cloneDeep(this[prop]);
         }
+      } else if (
+        prop &&
+        prop === 'type' &&
+        this.sliceName &&
+        this.id.slice(0, this.id.lastIndexOf(':')).endsWith('.extension')
```

Comment:
I'm getting regressions on this PR, and they appear to be tied to the fact that this check also catches "inline" extensions. Here is an example that can be used to recreate what I'm seeing:
```
Extension: Inline
* extension contains
    volume 1..1 MS
```
On this branch, you are going to get something like:
```json
{
  "id": "Extension.extension:volume",
  "path": "Extension.extension",
  "sliceName": "volume",
  "min": 1,
  "max": "1",
  "type": [
    {
      "code": "Extension"
    }
  ],
  "mustSupport": true
}
```
where that `type` information is really not needed, since the type is unchanged. On `master`, you get no such type info.

Reply:
This is a good catch! Addressed in commit `1a4f635`